sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/models/ShareFileItemProperties.java

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.file.share.models;
+
+import java.time.OffsetDateTime;
+
+/**
+ * Share item properties for items on a listing operation.
+ */
+public class ShareFileItemProperties {
+    private final OffsetDateTime createdOn;
+    private final OffsetDateTime lastAccessedOn;
+    private final OffsetDateTime lastWrittenOn;
+    private final OffsetDateTime changedOn;
+    private final OffsetDateTime lastModified;
+    private final String eTag;
+
+    /**
+     * Creates an instance of share item properties.
+     *
+     * @param createdOn Datetime the item was created.
+     * @param lastAccessedOn Datetime the item was last accessed.
+     * @param lastWrittenOn Datetime the item was last written.
+     * @param changedOn Datetime the item was last changed.
+     * @param lastModified Datetime the item was last modified.
+     * @param eTag ETag of the item.
+     */
+    public ShareFileItemProperties(OffsetDateTime createdOn, OffsetDateTime lastAccessedOn,
```

Comment:
If this is new type then I'd recommend to convert it into interface and have actual implementation in the "implementation" package. This will help us manage growing model without worry about breaking ctors.

Let's do this for all immutable models we add.